R/database.R

```diff
@@ -747,836 +870,951 @@ get_combined_sv = function(min_vaf=0,
+  all_counts = left_join(eg, ashm_counted) %>% 
+    mutate(n = replace_na(n, 0)) %>% 
+    unique() #not sure where the duplicates are coming from but its annoying
+
+  all_counts_wide = pivot_wider(all_counts, id_cols = sample_id, names_from = region_name, values_from = n) %>%
+    column_to_rownames(var = "sample_id")
 
-  all_counts_wide = pivot_wider(all_counts,id_cols = sample_id,names_from=region_name,values_from=n) %>%
-    column_to_rownames(var="sample_id")
   return(all_counts_wide)
 }
 
-#' Get all somatic mutations for a given gene or list of genes and optionally restrict to coding variants
+
+#' Get all somatic mutations for a given gene or list of genes and optionally restrict to coding variants.
 #'
-#' @param gene_symbol Character vector of gene symbols
-#' @param coding_only Logical parameter indicating whether to return only coding mutations. Default is FALSE
-#' @param include_silent Logical parameter indicating whether to include siment mutations into coding mutations. Default is TRUE
-#' @param rename_splice_region Logical parameter indicating whether to rename splice regions. Default is TRUE
+#' @param gene_symbol Character vector of gene symbols.
+#' @param coding_only Logical parameter indicating whether to return only coding mutations. Default is FALSE.
+#' @param include_silent Logical parameter indicating whether to include siment mutations into coding mutations. Default is TRUE.
+#' @param rename_splice_region Logical parameter indicating whether to rename splice regions. Default is TRUE.
 #'
-#' @return MAF-format data frame of mutations in query gene
+#' @return MAF-format data frame of mutations in query gene.
 #' @export
 #' @import tidyverse DBI RMariaDB dbplyr
 #'
 #' @examples
 #' #basic usage
-#' get_ssm_by_gene(gene_symbol=c("EZH2"),coding_only=TRUE)
-get_ssm_by_gene = function(gene_symbol,coding_only=FALSE,include_silent=TRUE,rename_splice_region=TRUE){
+#' ssm_ezh2 = get_ssm_by_gene(gene_symbol=c("EZH2"),coding_only=TRUE)
+#'
+get_ssm_by_gene = function(gene_symbol,
+                           coding_only = FALSE,
+                           include_silent = TRUE,
+                           rename_splice_region = TRUE){
+
   table_name = config::get("results_tables")$ssm
-  db=config::get("database_name")
-  coding_class = c("Frame_Shift_Del",
-                   "Frame_Shift_Ins",
-                   "In_Frame_Del",
-                   "In_Frame_Ins",
-                   "Missense_Mutation",
-                   "Nonsense_Mutation",
-                   "Nonstop_Mutation",
-                   "Silent",
-                   "Splice_Region",
-                   "Splice_Site",
-                   "Targeted_Region",
-                   "Translation_Start_Site")
+  db = config::get("database_name")
   if(!include_silent){
-    coding_class=coding_class[coding_class != "Silent"]
+    coding_class = coding_class[coding_class != "Silent"]
```

Comment:
This doesn't clobber the global variable of the same name, does it?

Reply:
If you create a variable with the same name inside a function, this variable will be local, and can only be used inside the function. The global variable with the same name will remain as it was, global and with the original value. I've also double-checked this, and it's true.